### DISCUSSION TITLE
Add Support for Passkeys and Hardware Security Keys (like YubiKeys)

### DISCUSSION THREAD

**Discussion by iAnonymous3000** [ente-io/ente#592] [Enhancements]
I would like to request the addition of support for passkeys and hardware security keys, such as YubiKeys, for enhancing the security of our application. This feature would provide users with more secure authentication options.

### Rationale
Passkeys and hardware security keys offer a more secure form of two-factor authentication (2FA) and can significantly reduce the risk of unauthorized access.

### Suggested Implementation
- **Passkey Support**: Implement passkey-based authentication to allow users to log in without needing to remember complex passwords, leveraging platform authenticators like biometrics or security keys.
- **Hardware Security Key Integration**: Allow users to register their YubiKeys (or other hardware security keys) as a method for two-factor authentication. This could be integrated through the WebAuthn API to provide a standardized approach to security key authentication.

### Use Cases
- **Enhanced Security**: Users with high-security needs, such as those handling sensitive data, would benefit from the added security layer.
- **Accessibility**: Users who have difficulty managing multiple complex passwords would find passkeys a more convenient and secure alternative.
- **Compliance**: Organizations requiring stringent security measures could enforce hardware key authentication for access control.

### Additional Context
Many leading platforms and services are moving towards passwordless authentication methods to enhance security and user experience. Implementing passkey and hardware security key support would not only improve security but also position our application as a leader in user data protection.

I believe this feature would be a significant step forward in our commitment to user security and privacy. I look forward to hearing your thoughts and any additional considerations for implementing this feature.

**Comment by mnvr** [comment]
Hey @iAnonymous3000, this is already being worked on! Expect more news soon.

Related discussion:
* https://github.com/ente-io/ente/discussions/502

**Comment by windmueller** [comment]
Please do not limit this to Yubikeys. U2F is an open standard supported by many vendors.

**Reply by iAnonymous3000** [reply]
Agreed! Not to worry, I think @mnvr means all hardware security keys here. There are many other options too, like Google Titan Security Keys, Nitrokeys, Solokeys, etc.

**Reply by mnvr** [reply]
Right, I was using Yubikeys as a generic term, apologies for the confusion. The current plan is to have support all standard hardware keys, not specific to any vendor.

**Comment by mnvr** [comment]
This is done now - Passkey support, including for Yubikey and other hardware keys, has been added. For more details, please see https://ente.io/blog/introducing-passkeys-on-ente/.
